language/golang/language_test.go

```diff
@@ -178,3 +178,97 @@ func TestLanguageExecute(t *testing.T) {
 		})
 	})
 }
+
+func TestMistakes(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Language *Language
+
+		RepositoryPath string
+
+		ExpectedMistakes []string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			temporaryPath := t.TempDir()
+			repositoryPath := filepath.Join(temporaryPath, filepath.Base(tc.RepositoryPath))
+			require.NoError(t, osutil.CopyTree(tc.RepositoryPath, repositoryPath))
+
+			buffer, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Log(buffer.String())
+				}
+			}()
+
+			if tc.Language == nil {
+				tc.Language = &Language{}
+			}
+			actualMistakes, actualErr := tc.Language.Mistakes(logger, repositoryPath)
+			require.NoError(t, actualErr)
+
+			assert.Equal(t, tc.ExpectedMistakes, actualMistakes)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Function without opening bracket",
+
+		RepositoryPath: filepath.Join("..", "..", "testdata", "golang", "mistakes"),
+
+		ExpectedMistakes: []string{
+			"." + string(os.PathSeparator) + "functionWithoutOpeningBracket.go" + ":4:6: syntax error: non-declaration statement outside function body",
```

Comment:
Maybe  `filepath.Join(".", "functionWithoutOpeningBracket.go")+":4...`

Reply:
To clarify it is about the file.

Reply:
See https://github.com/symflower/eval-dev-quality/pull/162#discussion_r1629091363